"NX Cannot find module '@nrwl/nx-linux-arm64-gnu'" when running angular universal in a node:18 docker

### Current Behavior

I just migrated using `nx migrate latest`
```
angular packages: 15.1.1 => 15.2.1

@nrwl/angular: 15.7.2 => 15.8.5
@nrwl/webpack: 15.7.2 => 15.8.5
```

I'm running angular universal in development mode and start it in a docker `node:18`

Now I get the following error:
`  >  NX   Cannot find module '@nrwl/nx-linux-arm64-gnu'`

As described in a similar issue, https://github.com/nrwl/nx/issues/8969
I tried to delete my `node_modules` and clear my cache.

It seems like my application works if I run my nx command outside docker:
`nx run my-app:serve-ssr`

When starting my docker image, it throws the error above.

If i check my package-lock.json the package is listed under "optionalDependencies":
`"@nrwl/nx-linux-arm64-gnu": "15.8.5"`

### Expected Behavior

After the migration it may should have installed the missing module `@nrwl/nx-linux-arm64-gnu` and run.

### GitHub Repo

_No response_

### Steps to Reproduce

1.
# Docker Image:
```
FROM node:18

RUN mkdir -p /app
WORKDIR /app/

COPY package*.json ./
COPY decorate-angular-cli.js ./decorate-angular-cli.js
RUN npm install -g @angular/cli@15 @angular-devkit/build-angular@15

RUN nx run my-app:serve-ssr
```



### Nx Report

```shell
Node : 18.14.2
   OS   : darwin arm64
   npm  : 9.5.0
   
   nx                      : 15.8.5
   @nrwl/js                : 15.8.5
   @nrwl/jest              : 15.8.5
   @nrwl/linter            : 15.8.5
   @nrwl/workspace         : 15.8.5
   @nrwl/angular           : 15.8.5
   @nrwl/cli               : 15.8.5
   @nrwl/cypress           : 15.8.5
   @nrwl/devkit            : 15.8.5
   @nrwl/eslint-plugin-nx  : 15.8.5
   @nrwl/storybook         : 15.8.5
   @nrwl/tao               : 15.8.5
   @nrwl/webpack           : 15.8.5
   typescript              : 4.9.5
   ---------------------------------------
   Community plugins:
   @nguniversal/express-engine : 15.2.0
   apollo-angular              : 4.2.1
   ng2-charts                  : 4.1.1
   @nguniversal/builders       : 15.2.0
   @storybook/angular          : 6.5.15
```


### Failure Logs

_No response_

### Additional Information

_No response_

@Cammisuli I'm facing with the same issue. My docker file content is the following:

```
# Stage 1
FROM node:latest as build

WORKDIR /usr/local/app
COPY ./ /usr/local/app/

RUN npm ci

# ARG NX_NON_NATIVE_HASHER=true
RUN npm run build -- ogs

# Stage 2
FROM nginx:latest

COPY --from=build /usr/local/app/dist/apps/my-app /usr/share/nginx/html
COPY nginx.conf /etc/nginx/conf.d/default.conf

EXPOSE 80
```
With this config I get `Cannot find module '@nrwl/nx-linux-x64-gnu'`, but as it can be seen I'm running `npm ci` which will remove the copied version of `node_modules` folder and install them again on the container.
The only workaround for now is to uncomment the `# ARG NX_NON_NATIVE_HASHER=true` line.

--- a/docs/shared/installation/troubleshoot-installation.md
+++ b/docs/shared/installation/troubleshoot-installation.md
@@ -12,8 +12,11 @@ Some npm users are experiencing errors like the following:
 NX   Cannot find module '@nrwl/nx-linux-x64-gnu'
 ```
 
-This happens because the package-lock.json file was not correctly updated by npm, and missed optional dependencies used by Nx.
-You can read more about this [issue on the npm repository.](https://github.com/npm/cli/issues/4828)
+There are two reasons why this could potentially happen:
+
+1. Running your install command with `--no-optional` (or the relative flag in yarn, pnpm, etc)
+2. The package-lock.json file was not correctly updated by npm, and missed optional dependencies used by Nx.
+   You can read more about this [issue on the npm repository.](https://github.com/npm/cli/issues/4828)
 
 {% callout type="note" title="Updating Nx" %}
 When updating Nx that is already on 15.8, the package-lock.json should continue to be updated properly with all the proper optional dependencies.
@@ -21,7 +24,8 @@ When updating Nx that is already on 15.8, the package-lock.json should continue 
 
 ### How to fix
 
-Deleting your node_modules and `package-lock.json` and re-running `npm i` should make sure that everything is created properly in the newly generated `package-lock.json`.
+1. If you are running your install command with `--no-optional`, try again without the flag.
+2. Delete your node_modules and `package-lock.json` and re-run `npm i`. This should have the `package-lock.json` file updated properly.
 
 ### Supported native module platforms
 

